xbmc/interfaces/legacy/Dialog.cpp

```diff
@@ -85,6 +88,27 @@ namespace XBMCAddon
       return pDialog->IsConfirmed();
     }
 
+    bool Dialog::videoinfo(const ListItem* item)
+    {
+      const AddonClass::Ref<xbmcgui::ListItem> listitem(item);
+      auto fileItem = new CFileItem(*listitem->item);
```

Comment:
Leaks. thou shalt not use new in c++ you know;) Something like this should suffice:

```
bool Dialog::videoinfo(const ListItem& item)
{
  if (!item.item->HasVideoInfoTag())
    return false;
  CGUIDialogVideoInfo::ShowFor(*item.item);
  return true;
}
```